src/test_server.py

```diff
@@ -1,79 +1,117 @@
 # -*- coding: utf-8 -*-
 """Test file performs tests on client.py and server.py."""
 from __future__ import unicode_literals
-from server import response_ok, response_error, parse_message
 import pytest
+try:
+    from http.client import HTTPException
+except ImportError:
+    from httplib import HTTPException
 
 CRLF = '\r\n'
 
 
-def test_parse_message_good(valid_200_response):
-    """Testing parse_message with a valid_200_response"""
-    header_lines, body = parse_message(valid_200_response)
-    print('header lines: {}'.format(header_lines))
-    print('body: {}'.format(body))
-    assert 1 == 1
+# --------------------------------------------------------------------
+# Start parse_message tests
+# --------------------------------------------------------------------
+
+
+def test_parse_request_no_clrfclrf(invalid_request):
+    """Test checks for no CLRF + CLRF."""
+    from server import parse_request
+    test_header = (b"GETTTP/1.1\r\n"
+                   b"Host: 127.0.0.1:5000\r\n"
+                   b"Connection: keep-alive\r\n"
+                   b"Accept-Language: en-US,en;q=0.8\r\n"
+                   b"This is a sample message body!")
+    with pytest.raises(HTTPException) as excinfo:
+        parse_request(test_header)
+    assert '400 Bad Request' in str(excinfo)
+
+
+def test_parse_request_valid_get_http11(valid_get):
+    """Test that parse_request returns a valid GET path HTTP/1.1 request."""
+    from server import parse_request
+    assert parse_request(valid_get) == u'/path/file.html'
+
+
+def test_parse_request_bad_first_line():
+    """Test that parse_request returns exception for bad first line."""
+    from server import parse_request
+    test_header = (b"GETTTP/1.1\r\n"
+                   b"Host: 127.0.0.1:5000\r\n"
+                   b"Connection: keep-alive\r\n"
+                   b"Accept-Language: en-US,en;q=0.8\r\n\r\n"
+                   b"This is a sample message body!")
+    with pytest.raises(HTTPException) as excinfo:
+        parse_request(test_header)
+    assert '400 Bad Request' in str(excinfo)
+
+
+def test_parse_requset_bad_first_line_two():
```

Comment:
I'd prefer test names that explain why they are different from the other `bad_first_line` test(s)